src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/CommandLineArguments.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices
+{
+    public class CommandLineArguments
```

Comment:
We're not opening up `CommandLineArguments`, because that quickly snowballed into opening up way more internal stuff than we'd like.  I'm only opening up the constructors to the structs that this new type contains in its `IEnumerable<>` properties, and the properties on this class are the only things that were being used, anyways.